presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -847,13 +847,18 @@ public boolean supportsMissingColumnsOnInsert(Session session, TableHandle table
     }
 
     @Override
-    public InsertTableHandle beginRefreshMaterializedView(Session session, TableHandle tableHandle)
+    public InsertTableHandle beginRefreshMaterializedView(Session session, TableHandle tableHandle, List<TableHandle> sourceTableHandles)
     {
         CatalogName catalogName = tableHandle.getCatalogName();
         CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, catalogName);
         ConnectorMetadata metadata = catalogMetadata.getMetadata();
         ConnectorTransactionHandle transactionHandle = catalogMetadata.getTransactionHandleFor(catalogName);
-        ConnectorInsertTableHandle handle = metadata.beginRefreshMaterializedView(session.toConnectorSession(catalogName), tableHandle.getConnectorHandle());
+        List<ConnectorTableHandle> sourceConnectorHandles = new ArrayList<>();
```

Comment:
This type of operation can be done with a stream (though we'll need to change it anyway so we filter out tables from other connectors)

Reply:
Please see my response to cross-connector materialized views above. Keeping in line with that, I did not make any changes here. I rewrote the loop to use streams though. (here and in FinishRefreshMaterializedView call).